test/performance/kshelper.py

```diff
@@ -106,72 +115,142 @@ def get_crd_time(self, grp, v, pl, name=None, check=False):
                         op_type = "subresource"
 
                         if operation == "Update":
-                            if manager in dict_manager_list:
-                                if op_type in managedFields[j]:
-                                    update_time = managedFields[j]["time"]
-                                    op_manager = manager
-                                    break
+                            if manager in manager_list:
+                                if pl == "manifestworks": 
+                                   # Handle manifestwork objects differently: extract the timestamp when transport-controller updates a manifestwork object 
+                                   if manager == "transport-controller":
+                                      update_timestamp = managedFields[j]["time"]
+                                      op_manager = manager
+                                      break
+                                   else:
+                                     continue
+                                elif op_type in managedFields[j]:
+                                     update_timestamp = managedFields[j]["time"]
+                                     op_manager = manager
+                                     break
                         else:
-                            continue 
-                    fout.write(name + "\t" + creation_time + "\t" + update_time + "\t" + status_condition + "\t" + op_manager + "\n")
-            fout.close()
+                            continue
+
+                    output=obj_name + "\t" + creation_timestamp + "\t" + update_timestamp + "\t" + status_condition + "\t" + op_manager
+                    if name != None and single == True:
+                       return output
+                    else:
+                       fout.write(output+ "\n")
+
+            if name == None or single == False:
+               fout.flush()
+               fout.close()
+            return None
+
         except ApiException as e:
             print("Exception when calling client->CustomObjectsApi(): %s\n" % e)
 
 
-    def get_pod_time(self, ns, name=None):
+    def get_obj_time(self, fname, kind, ns, name=None):
         try:
+            kind = kind.lower()
             if name != None:
-                api_response = self.core_v1_api.list_namespaced_pod(ns, field_selector='metadata.name=' + name)
-            else: 
-                api_response = self.core_v1_api.list_namespaced_pod(ns)
+                if kind == "pod" or kind == "pods":
+                   api_response = self.core_v1_api.list_namespaced_pod(ns, field_selector='metadata.name=' + name)
```

Comment:
Why use `list` instead of `get` when the name is known?

Reply:
Fixed! using `get` instead to improve performance